src/main/java/Sinep/Sinep.java

```diff
@@ -0,0 +1,322 @@
+        int taskIndex = Integer.parseInt(input.substring(7)) - 1;
+        if (taskIndex >= taskList.size()) {
+            throw new InvalidTaskIndexException();
+        }
+        Task markingTask = taskList.get(taskIndex);
+        markingTask.unmarkAsDone();
+        System.out.println(line + nl + "Got it! Sinep.Task " + (taskIndex + 1) + " marked as undone:");
+        System.out.println(markingTask.getStatusIcon() + " " + markingTask.description + nl + line);
+    }
+
+
+    public static void addTodo(String input) throws InvalidCommandMessageException {
+        if (Objects.equals(input, "todo")) {
+            throw new InvalidCommandMessageException();
+        }
+        String actualDescription = input.replace("todo ", "");
+        Todo newTodo = new Todo(actualDescription);
+        taskList.add(newTodo);
+        System.out.println(line + nl + "Added to task list:" + nl + newTodo);
+        System.out.println("Now you have " + taskList.size() + " tasks in the list." + nl + line);
+    }
+
+    public static void addDeadline(String input) throws InvalidCommandMessageException{
+        boolean containsDeadline = input.contains("/by");
+        if (Objects.equals(input, "deadline") || !containsDeadline) {
+            throw new InvalidCommandMessageException();
+        }
+        String actualDescription = input.replace("deadline ", "");
+        String[] inputParts = actualDescription.split("/by", 2);
+        if (Objects.equals(inputParts[0], "") || Objects.equals(inputParts[1], "")) {
+            throw new InvalidCommandMessageException();
+        }
+        Deadline newDeadline = new Deadline(actualDescription);
+        taskList.add(newDeadline);
+        String descriptionToPrint = newDeadline.toString();
+        System.out.println(line + nl + "Added to task list:" + nl + descriptionToPrint);
+        System.out.println("Now you have " + taskList.size() + " tasks in the list." + nl + line);
+    }
+
+    public static void addEvent(String input) throws InvalidCommandMessageException{
+        boolean containStart = input.contains("/from");
+        boolean containEnd = input.contains("/to");
+        if (Objects.equals(input, "event") || !containStart || !containEnd) {
+            throw new InvalidCommandMessageException();
+        }
+        String actualDescription = input.replace("event ", "");
+        String[] inputParts = actualDescription.split("/from|/to", 3);
+        if (Objects.equals(inputParts[0], "") || Objects.equals(inputParts[1], "") || Objects.equals(inputParts[2], "")) {
+            throw new InvalidCommandMessageException();
+        }
+        Event newEvent = new Event(actualDescription);
+        taskList.add(newEvent);
+        String descriptionToPrint = newEvent.toString();
+        System.out.println(line + nl + "Added to task list:" + nl + descriptionToPrint);
+        System.out.println("Now you have " + taskList.size() + " tasks in the list." + nl + line);
+    }
+
```

Comment:
In general, I believe the `Sinep` class has a lot of methods, that are not necessarily related to it, and these may be abstracted to another Class such as `TaskManager` or `TaskList`.